pilot/pkg/model/validation.go

```diff
@@ -2030,7 +2030,9 @@ func validateCORSPolicy(policy *networking.CorsPolicy) (errs error) {
 		return
 	}
 
-	// TODO: additional validation for AllowOrigin?
+	for _, host := range policy.AllowOrigin {
+		errs = appendErrors(errs, ValidateWildcardDomain(host))
```

Comment:
I think this is too strict, xref: https://stackoverflow.com/questions/19541547/does-cross-origin-resource-sharingcors-differentiate-between-http-and-https/19542686#19542686

Reply:
Made changes accordingly.

Reply:
I am not familiar with cors, better have a look at official doc about `Access-Control-Allow-Origin`.

Reply:
Reading the reference `https://www.w3.org/TR/cors/#access-control-allow-origin-response-header` it seems to me that "*" is the only allowed host name containing a wildcard. Updating the PR accordingly.